trieste/acquisition/optimizer.py

```diff
@@ -35,24 +34,81 @@
-@singledispatch
-def optimize(space: Box | DiscreteSearchSpace, target_func: AcquisitionFunction) -> TensorType:
+def automatic_optimizer_selector(
+    space: DiscreteSearchSpace, target_func: AcquisitionFunction
+) -> TensorType:
     """
+    A wrapper around our :const:`AcquisitionOptimizer`s. This class performs
+    an :const:`AcquisitionOptimizer` appropriate for the
+    problem's :class:`~trieste.space.SearchSpace`.
+
     :param space: The space of points over which to search, for points with shape [D].
-    :param target_func: The function to maximise, with input shape [..., D] and output shape
+    :param target_func: The function to maximise, with input shape [..., 1, D] and output shape
         [..., 1].
-    :return: The point in ``space`` that maximises ``target_func``, with shape [1, D].
+    :return: The batch of points in ``space`` that maximises ``target_func``, with shape [1, D].
+    """
+
+    if isinstance(space, Box):
+        optimizer = optimize_continuous
+    elif isinstance(space, DiscreteSearchSpace):
+        optimizer = optimize_discrete
+    else:
+        raise NotImplementedError(
+            """ No optimizer currentely supports acquisition function
+            maximisation over this search space"""
+        )
+
+    return optimizer(space, target_func)
+
+
+def batchify(
+    batch_size_one_optimizer: AcquisitionOptimizer,
+    batch_size: int,
+) -> AcquisitionOptimizer[SP]:
     """
+    A wrapper around our :const:`AcquisitionOptimizer`s. This class wraps a
+    :const:`AcquisitionOptimizer` to allow it to optimize batch acquisition functions.
+
+    :param batch_size_one_optimizer: An optimizer that returns only batch size one, i.e. produces a
+        single point with shape [1, D].
+    :param batch_size: The number of points in the batch.
+    :return: An :const:`AcquisitionOptimizer` that will provide a batch of points with shape [B, D].
+    """
+    tf.debugging.assert_positive(batch_size)
+
+    def optimizer(search_space: SP, f: AcquisitionFunction) -> TensorType:
+        expanded_search_space = search_space ** batch_size  # points have shape [B * D]
+
+        def target_func_with_vectorized_inputs(
+            x: TensorType,
+        ) -> TensorType:  # [..., 1, B * D] -> [..., 1]
+            return f(tf.reshape(x, x.shape[:-2].as_list() + [batch_size, -1]))
+
+        vectorized_points = batch_size_one_optimizer(  # [1, B * D]
+            expanded_search_space, target_func_with_vectorized_inputs
+        )
+        return tf.reshape(vectorized_points, [batch_size, -1])  # [B, D]
+
+    return optimizer
 
 
-@optimize.register
-def _discrete_space(space: DiscreteSearchSpace, target_func: AcquisitionFunction) -> TensorType:
-    target_func_values = target_func(space.points)
+def optimize_discrete(space: DiscreteSearchSpace, target_func: AcquisitionFunction) -> TensorType:
+    """
+    An :const:`AcquisitionOptimizer` for :const:'DiscreteSearchSpace' spaces and
+    batches of size of 1.
```

Comment:
```suggestion
    An :const:`AcquisitionOptimizer` for :class:'DiscreteSearchSpace' spaces and
    batches of size of 1.
```
`:const:` is for constants, `:class:` is for classes